sql-db/sql-app-oracle/src/test/java/io/quarkus/ts/sqldb/sqlapp/oracle/AbstractSqlDatabaseIT.java

```diff
@@ -0,0 +1,197 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import io.restassured.http.ContentType;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public abstract class AbstractSqlDatabaseIT {
+
+    private static final int EXPECTED_SIZE = 7;
+
+    private static final int VALID_ID = 8;
+    private static final int INVALID_ID = 999;
+
+    @Test
+    @Order(1)
+    public void getAll() {
+        given()
+                .get("/book")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body("", hasSize(EXPECTED_SIZE));
+    }
+
+    @Test
+    @Order(2)
+    public void get() {
+        given()
+                .get("/book/7")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body("title", equalTo("Perdido Street Station"))
+                .body("author", equalTo("China Miéville"));
+    }
+
+    @Test
+    @Order(3)
+    public void create() {
+        Book book = new Book();
+        book.title = "Neuromancer";
+        book.author = "William Gibson";
+
+        given()
+                .contentType(ContentType.JSON)
+                .body(book)
+                .post("/book")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED)
+                .body("id", equalTo(VALID_ID))
+                .body("title", equalTo("Neuromancer"))
+                .body("author", equalTo("William Gibson"));
+
+        given()
+                .get("/book/8")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body("title", equalTo("Neuromancer"))
+                .body("author", equalTo("William Gibson"));
+    }
+
+    @Test
+    @Order(4)
+    public void createInvalidPayload() {
+        given()
+                .contentType(ContentType.TEXT)
+                .body("")
+                .post("/book")
+                .then()
+                .statusCode(HttpStatus.SC_UNSUPPORTED_MEDIA_TYPE)
+                .body("code", equalTo(HttpStatus.SC_UNSUPPORTED_MEDIA_TYPE));
+    }
+
+    @Test
+    @Order(5)
+    public void createBadPayload() {
+        Book book = new Book();
+        book.id = Long.valueOf(INVALID_ID);
+        book.title = "foo";
+        book.author = "bar";
+
+        given()
+                .contentType(ContentType.JSON)
+                .body(book)
+                .post("/book")
+                .then()
+                .statusCode(HttpStatus.SC_UNPROCESSABLE_ENTITY)
+                .body("code", equalTo(HttpStatus.SC_UNPROCESSABLE_ENTITY))
+                .body("error", equalTo("unexpected ID in request"));
+    }
+
+    @Test
+    @Order(6)
+    public void update() {
+        Book book = new Book();
+        book.id = Long.valueOf(VALID_ID);
+        book.title = "Schismatrix";
+        book.author = "Bruce Sterling";
+
+        given()
+                .contentType(ContentType.JSON)
+                .body(book)
+                .put("/book/8")
```

Comment:
Could this(as well as similar code on line 137) be replaced with `.get("/book/" + VALID_ID)`? Right now, one need to go to the start of the file to understand, what is happening in these two methods.

Reply:
So the point is to use the `VALID_ID` constant (and possibly other constants) wherever possible, to avoid confusion, right?
If so, I think I can replace it in a few more occurences.
However, the `put` method is IMO correct here.

Reply:
Yes, "put" is definitely belongs here :)

No, you don't need to use VALID_ID constant  everywhere, just use(or don't use) it consistently inside the same method. When we create book with id=VALID_ID, but access it through "books/8" URL, it's not really clear, that these ids are connected.